provider/repodata/data/json/props/bidiauxiliaryprops@1/und.json

```diff
@@ -0,0 +1,6633 @@
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
+      {
+        "mirroring_glyph": "\u0000",
+        "mirrored": false,
+        "paired_bracket_type": 0
+      },
```

Comment:
Observation: the great majority of these entries are all the same. It would be nice if CPT gave us better compression.

Reply:
CPT Builder compaction isn't a compression algorithm like gzip. It just tries to find ways in which the value in the last hop from the index array to the data array can be reused without loss of correctness. The last 6 bits of the code point's bit field are used to advance from the offset index in the data array given by the index array. The 2^6=64 values starting from that offset are called a "block", and so that's the level of granularity (more or less) that compaction can happen).

The levels of the trie depend on the number and size of partitions of the code point's bit field, and those decisions all a tradeoff between space and time. The ones for this iteration of CodePointTrie also work well for UTF-8 encodings (ex: the 6-bit size allows you to skip some of the work of reconstructing the code point from the UTF-8 trail bytes). I don't know if that answered your question, but hopefully it was relevant.

@markusicu